idaes/apps/grid_integration/coordinator.py

```diff
@@ -0,0 +1,879 @@
+
+        current_ruc_dispatch_dicts = [
+            simulator.data_manager.ruc_market_active.thermal_gen_cleared_DA,
+            simulator.data_manager.ruc_market_active.renewable_gen_cleared_DA,
+            simulator.data_manager.ruc_market_active.virtual_gen_cleared_DA,
+        ]
+
+        tracking_horizon = len(self.projection_tracker.time_set)
+
+        market_signals = self._assemble_sced_tracking_market_signals(
+            gen_name=gen_name,
+            hour=hour,
+            sced_dispatch=None,
+            tracking_horizon=tracking_horizon,
+            current_ruc_dispatch_dicts=current_ruc_dispatch_dicts,
+            next_ruc_dispatch_dicts=None,
+        )
+        return market_signals
+
+    def project_tracking_trajectory(self, options, simulator, ruc_hour):
+
+        """
+        This function projects the full power dispatch trajectory from the
+        tracking model so we can use it to update the bidding model, i.e. advance
+        the time for the bidding model.
+
+        Arguments:
+            options: Prescient options.
+
+            simulator: Prescient simulator.
+
+            ruc_hour: the hour RUC is being solved
+
+        Returns:
+            full_projected_trajectory: the full projected power dispatch trajectory.
+        """
+
+        current_date = simulator.time_manager.current_time.date
+        current_hour = simulator.time_manager.current_time.hour
+
+        self._clone_tracking_model()
+
+        for hour in range(ruc_hour, 24):
+
+            # assemble market_signals
+            market_signals = self.assemble_project_tracking_signal(
+                options=options, simulator=simulator, hour=hour
+            )
+            # solve tracking
+            self.projection_tracker.track_market_dispatch(
+                market_dispatch=market_signals, date=current_date, hour=current_hour
+            )
+
+        # merge the trajectory
+        full_projected_trajectory = {}
+        for stat in self.tracker.daily_stats:
+            full_projected_trajectory[stat] = self.tracker.daily_stats.get(
+                stat
+            ) + self.projection_tracker.daily_stats.get(stat)
+
+        # clear the projection stats
+        self.projection_tracker.daily_stats = None
+
+        return full_projected_trajectory
+
+    def _clone_tracking_model(self):
+        """
+        Clone the model in tracker and replace that of projection tracker. In this
+        way, tracker and projection tracker have the same states before projection.
+
+        Arguments:
+            None
+
+        Returns:
+            None
+        """
+        # self.projection_tracker.model = self.tracker.model.clone()
+
+        objects_list = [pyo.Var, pyo.Param]
+        for obj in objects_list:
+            for tracker_obj, proj_tracker_obj in zip(
+                self.tracker.model.component_objects(obj),
+                self.projection_tracker.model.component_objects(obj),
+            ):
```

Comment:
I worry a bit about using `zip` on `component_objects` like this. I suppose it'll be okay if they're literally the same Pyomo model...

@jsiirola thoughts?

Reply:
@jsiirola Thoughts?

Reply:
This is really asking for trouble.  You are counting on the declaration order and all sets to match between the two models.  It might be sufficient to set `sort=SortComponents.alphabetizeComponentAndIndex` when calling `component_objects`.  You should still verify that the two component names match, though...